crypto/evp/names.c

```diff
@@ -251,3 +280,46 @@ void EVP_MAC_do_all_sorted(void (*fn)
     OBJ_NAME_do_all_sorted(OBJ_NAME_TYPE_MAC_METH, do_all_mac_fn, &dc);
 }
 
+struct doall_kdf {
+    void *arg;
+    void (*fn) (const EVP_KDF *kdf,
+                const char *from, const char *to, void *arg);
+};
+
+static void do_all_kdf_fn(const OBJ_NAME *nm, void *arg)
+{
+    struct doall_kdf *dc = arg;
+
+    if (nm->alias)
+        dc->fn(NULL, nm->name, nm->data, dc->arg);
+    else
+        dc->fn((const EVP_KDF *)nm->data, nm->name, NULL, dc->arg);
+}
+
+void EVP_KDF_do_all(void (*fn)
+                    (const EVP_KDF *kdf, const char *from, const char *to,
+                     void *x), void *arg)
+{
+    struct doall_kdf dc;
+
+    /* Ignore errors */
+    OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_KDFS, NULL);
+
+    dc.fn = fn;
+    dc.arg = arg;
+    OBJ_NAME_do_all(OBJ_NAME_TYPE_KDF_METH, do_all_kdf_fn, &dc);
+}
+
+void EVP_KDF_do_all_sorted(void (*fn)
```

Comment:
Similarly here. Why do we need this?

Reply:
There are also EVP_cipher and EVP_MD do_all functions.. Not sure how useful they are..

Reply:
removed